middlewares/authenticate.js

```diff
@@ -83,13 +84,31 @@ module.exports = async (req, res, next) => {
 
         // add user data to `req.userData` for further use
         req.userData = await users.fetchUser({ userId })
-
         return checkRestricted(req, res, next)
-      } else {
-        return res.boom.unauthorized('Unauthenticated User')
       }
-    } else {
-      return res.boom.unauthorized('Unauthenticated User')
     }
+
+    // If auth is not strictly required, just call the next handler.
+    if (!strict) {
+      return next()
+    }
+
+    return res.boom.unauthorized('Unauthenticated User')
   }
 }
+
+/**
+ * Check if authentication is present. If the request contains user auth, then add user details to the req object.
+ * If user is not logged in, just call the next handler.
+ * @param {Object} req
+ * @param {Object} res
+ * @param {function} next
+ */
+const maybeAuthenticate = async (req, res, next) => {
+  authenticate(req, res, next, false)
+}
```

Comment:
Are we sure this is doing a conditional check as described in the comments above?

Reply:
I tested it manually.

I'll write a test 😄

Reply:
Is maybeAuthenticate a good naming convention?

Reply:
Yes we should probably rename this

Reply:
@swarajpure @vinayak-trivedi could you suggest a better name for it.